__tests__/schema/events.ts

```diff
@@ -0,0 +1,361 @@
+        query events {
+          events(last: 150) {
+            nodes {
+              __typename
+              id
+            }
+          }
+        }
+      `,
+      client,
+      message: 'last must be smaller or equal 100',
+    })
+  })
+})
+
+test('User.eventsByUser returns events of this user', async () => {
+  const events = updateIds(
+    R.concat(
+      allEvents.map(R.assoc('actorId', user.id)),
+      allEvents.map(R.assoc('actorId', user.id + 1))
+    )
+  )
+  setupEvents(events)
+  global.server.use(createUuidHandler(user))
+
+  await assertSuccessfulGraphQLQuery({
+    query: gql`
+      query userEvents($id: Int) {
+        uuid(id: $id) {
+          ... on User {
+            eventsByUser {
+              nodes {
+                __typename
+                id
+              }
+            }
+          }
+        }
+      }
+    `,
+    variables: { id: user.id },
+    client,
+    data: {
+      uuid: {
+        eventsByUser: {
+          nodes: events.slice(0, allEvents.length).map(getTypenameAndId),
+        },
+      },
+    },
+  })
+})
+
+test('AbstractEntity.events returns events for this entity', async () => {
+  const events = updateIds(
+    R.concat(
+      allEvents.map(R.assoc('objectId', article.id)),
+      allEvents.map(R.assoc('objectId', article.id + 1))
+    )
+  )
+  setupEvents(events)
+  global.server.use(createUuidHandler(article))
+
+  await assertSuccessfulGraphQLQuery({
+    query: gql`
+      query articleEvents($id: Int) {
+        uuid(id: $id) {
+          ... on Article {
+            events {
+              nodes {
+                __typename
+                id
+              }
+            }
+          }
+        }
+      }
+    `,
+    variables: { id: article.id },
+    client,
+    data: {
+      uuid: {
+        events: {
+          nodes: events.slice(0, allEvents.length).map(getTypenameAndId),
+        },
+      },
+    },
+  })
+})
+
+function setupEvents(
+  events: Model<'AbstractNotificationEvent'>[],
+  options?: {
+    maxReturn?: number
+  }
+) {
+  const maxReturn = options?.maxReturn ?? events.length / 2 + 1
+
+  global.server.use(
+    createDatabaseLayerHandler<{ after?: number }>({
+      matchMessage: { type: 'EventsQuery' },
+      resolver(req, res, ctx) {
+        const { after } = req.body.payload
+
+        if (after == null) {
+          return res(ctx.json({ events: events.slice(0, maxReturn) }))
+        }
+
+        const index = events.findIndex((e) => e.id === after)
+
+        if (index < 0) return res(ctx.status(400))
+
+        const updatedEvents = events.slice(index + 1, index + 1 + maxReturn)
+
+        return res(ctx.json({ events: updatedEvents }))
+      },
+    })
+  )
+}
+
+function getTypenameAndId(event: Model<'AbstractNotificationEvent'>) {
+  return R.pick(['__typename', 'id'], event)
+}
+
+function updateIds(events: Model<'AbstractNotificationEvent'>[]) {
+  return events.map((event, id) => R.assoc('id', id + 1, event))
+}
```

Comment:
```suggestion
function assignSequentialIds(events: Model<'AbstractNotificationEvent'>[]) {
  return events.map((event, id) => R.assoc('id', id + 1, event))
}
```

if I understand the intention of this function correctly.